beacon-chain/state/state-native/v1/state_trie.go

```diff
@@ -195,11 +278,25 @@ func (b *BeaconState) HashTreeRoot(ctx context.Context) ([32]byte, error) {
 
 	b.lock.Lock()
 	defer b.lock.Unlock()
-	if err := b.initializeMerkleLayers(ctx); err != nil {
-		return [32]byte{}, err
+
+	if b.merkleLayers == nil || len(b.merkleLayers) == 0 {
+		fieldRoots, err := computeFieldRoots(ctx, b)
+		if err != nil {
+			return [32]byte{}, err
+		}
+		layers := stateutil.Merkleize(fieldRoots)
+		b.merkleLayers = layers
+		b.dirtyFields = make(map[types.FieldIndex]bool, params.BeaconConfig().BeaconStateFieldCount)
 	}
-	if err := b.recomputeDirtyFields(ctx); err != nil {
-		return [32]byte{}, err
+
+	for field := range b.dirtyFields {
```

Comment:
same thing here

Reply:
accident most likely

Reply:
Same thing here again, this is a copy of `recomputeDirtyFields`